### PR TITLE
AbsSystem Miscellany

### PR DESCRIPTION
**Pull request by profxj**
Several small changes including a bug fix or two,
mainly related to AbsSystem objects.

Also modified the splice algorithm for spectra.

Tests updated.
No new docs.

### CODE REVIEW

File: linetools/spectra/utils.py [hunk 1]
```diff
@@ -334,7 +340,7 @@ def rebin(spec, new_wv, do_sig=False, do_co=False, all=False, **kwargs):
     return newspec
 
 
-def rebin_to_rest(spec, zarr, dv, debug=False):
+def rebin_to_rest(spec, zarr, dv, debug=False, **kwargs):
```
Comment (ntejos) [inline linetools/spectra/utils.py]: Mention that `**kwargs` are passed down to `spec.rebin()` ?

File: linetools/isgm/abscomponent.py [hunk 2]
```diff
@@ -256,8 +256,8 @@ def add_absline(self, absline, tol=0.1*u.arcsec, chk_vel=True,
         # Now redshift/velocity
         if chk_vel:
             dz_toler = (1 + self.zcomp) * vtoler / c_kms  # Avoid Quantity for speed
-            zlim_line = (1 + absline.z) * absline.limits.vlim.to('km/s').value / c_kms
-            zlim_comp = (1+self.zcomp) * self.vlim.to('km/s').value / c_kms
+            zlim_line = absline.limits.zlim  # absline.z + (1 + absline.z) * absline.limits.vlim.to('km/s').value / c_kms
+            zlim_comp = self.zcomp + (1+self.zcomp) * self.vlim.to('km/s').value / c_kms
```
Comment (profxj) [inline linetools/isgm/abscomponent.py]: I prefer to do this calculation without units to keep it fast.

File: linetools/isgm/abssystem.py [hunk 3]
```diff
@@ -521,6 +521,38 @@ def to_dict(self):
...
+        """ Update vlim in the main or subsystems
+
+        Parameters
+        ----------
+        sub_system : str, optional
+          If provided, apply to given sub-system.  Only used in LLS so far
+        """
+        def get_vmnx(components):
+            sys_vlim_mks = self.vlim.to('km/s').value
+            zlim_sys = self.zabs + (1 + self.zabs) * (sys_vlim_mks / c_mks)
```
Comment (profxj) [inline linetools/isgm/abssystem.py]: Done

File: linetools/spectra/utils.py (hunk 1 above)
Comment (profxj) [inline linetools/spectra/utils.py]: Done.

**Comment by profxj** [comment]
Have addressed the requested changes.

**Pull request merged**